linkerd/meshtls/boring/src/client.rs

```diff
@@ -76,25 +81,36 @@ where
             let conn = connector.map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
             let config = conn
                 .configure()
-                .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
-            let io = tokio_boring::connect(config, server_name.as_str(), io)
+                .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?
+                // Switch off DNS san Verification based on the hostname
+                .verify_hostname(false);
+
+            tokio_boring::connect(config, server_name.as_str(), io)
                 .await
                 .map_err(|e| match e.as_io_error() {
                     // TODO(ver) boring should let us take ownership of the error directly.
                     Some(ioe) => io::Error::new(ioe.kind(), ioe.to_string()),
                     // XXX(ver) to use the boring error directly here we have to constraint the socket on Sync +
                     // std::fmt::Debug, which is a pain.
                     None => io::Error::new(io::ErrorKind::Other, "unexpected TLS handshake error"),
-                })?;
-
-            debug!(
-                tls = io.ssl().version_str(),
-                client.cert = ?io.ssl().certificate().and_then(super::fingerprint),
-                peer.cert = ?io.ssl().peer_certificate().as_deref().and_then(super::fingerprint),
-                alpn = ?io.ssl().selected_alpn_protocol(),
-                "Initiated TLS connection"
-            );
-            Ok(ClientIo(io))
+                }).and_then(|io| match io.ssl().peer_certificate() {
+                Some(ref cert) => {
+                    verify::verify_id(cert, &server_id)
+                        .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
+                    debug!(
+                        tls = io.ssl().version_str(),
+                        client.cert = ?io.ssl().certificate().and_then(super::fingerprint),
+                        peer.cert = ?io.ssl().peer_certificate().as_deref().and_then(super::fingerprint),
+                        alpn = ?io.ssl().selected_alpn_protocol(),
+                        "Initiated TLS connection"
+                    );
+                    Ok(ClientIo(io))
+                },
+                         None => Err(io::Error::new(
+                    io::ErrorKind::Other,
+                    "could not extract peer cert",
+                )),
+                })
```

Comment:
some weird formatting going on here...i think it should look more like this:
```suggestion
                }).and_then(|io| match io.ssl().peer_certificate() {
                    Some(ref cert) => {
                        verify::verify_id(cert, &server_id)
                            .map_err(|e| io::Error::new(io::ErrorKind::Other, e))?;
                        debug!(
                            tls = io.ssl().version_str(),
                            client.cert = ?io.ssl().certificate().and_then(super::fingerprint),
                            peer.cert = ?io.ssl().peer_certificate().as_deref().and_then(super::fingerprint),
                            alpn = ?io.ssl().selected_alpn_protocol(),
                            "Initiated TLS connection"
                        );
                        Ok(ClientIo(io))
                    },
                    None => Err(io::Error::new(
                        io::ErrorKind::Other,
                        "could not extract peer cert",
                    )),
                })
```

have we run `rustfmt` on this branch?